dockerv2.go

```diff
@@ -15,7 +17,16 @@ var dockerRegexs = map[string]string{
 
 var DockerRegex = regexp.MustCompile("^\\/v2\\/(.*\\/(tags|manifests|blobs)\\/.*|_catalog$)")
 
-func generateDockerv2URI(path string, rec record) (string, int) {
+func redirectDockerv2(w http.ResponseWriter, r *http.Request, rec record) error {
+	path := r.URL.Path
+	v2Regex, err := regexp.Compile(dockerRegexs["v2"])
+	if err != nil {
+		panic(err)
+	}
+	if v2Regex.Match([]byte(path)) {
```

Comment:
The v2Regex matches all docker specific regexes and therefore returns StatusOK for all requests. Breaking the other redirects.